Fixing test suite

Nek- opened the pull request:
Hello there, trying to help to move forward.

This PR removes the test suite for PHP 8.0 because the dev dependencies are just not compatible with PHP 8.0.

It updates Sylius version to 1.10 and fixes behat tests.

It also allow to update to Symfony 5.3 thanks to the removal of the usage of `symfony/event-dispatcher-contracts` in favor of `symfony/contracts`.

Work is split in many commits to help you review with ease.

This PR targets 2.x but I plan to make it also on 3.x branch. Please send feedbacks!

Feel free to add `hacktoberfest-accepted` to help me complete my hacktoberfest challenge 😊 .

Nek- committed:
fix: branch alias in composer.json

Nek- committed:
update: test suite to Sylius 1.10

This is the only way to test with PHP 8.x and Symfony 5.

Nek- committed:
fix: test suite is not compatible with PHP 8.0

loevgaard commented on composer.json:
```diff
@@ -24,7 +24,7 @@
         "symfony/dependency-injection": "^4.4 || ^5.0",
         "symfony/event-dispatcher": "^4.4 || ^5.0",
-        "symfony/event-dispatcher-contracts": "^1.1",
+        "symfony/contracts": "^v1.1.10 || ^2.4.0",
```
Why is this change needed?

Nek- replied:
This is a dependency of the package [symfony/symfony](https://packagist.org/packages/symfony/symfony) and stands as replacement for `symfony/event-dispatcher-contracts`

![image](https://user-images.githubusercontent.com/972456/138958512-c0c6864a-67c9-4675-a8a6-1e4d0bc0f284.png)

Maybe I missread the situation and using `symfony/event-dispatcher-contracts` with constraint `^1.0 || ^2.0` is fine. Composer was suggesting using `symfony/contracts` instead so I did...

Nek- replied:
https://github.com/Setono/SyliusAnalyticsPlugin/pull/57 was working great! I removed my change for that.

loevgaard commented:
Thank you for this, @Nek-. The v2 of this plugin will not be maintained a short period after v3 is released (at least only security fixes), so if you see any improvements on v3 we would appreciate this also :)

Nek- commented:
@loevgaard the v3 has some issues in common (ie test suite not compatible with PHP 8 because of unused dev dependencies or too old dependencies (like sylius 1.7)) this is why I'm looking into updating both versions. But there's also the tag bundle to update (I'll send some work tomorrow about it).

I see the test suite is not passing yet, be sure I will fix it!

Nek- committed:
feat: update changes required by psalm & cs

Nek- committed:
fix: behat tests in Symfony 5.3

https://github.com/symfony/symfony/issues/41672 is blocking to update
the behat test suite. This is why I decided to add this patch. It should
not have any specific impact but for the dev environment.

After the issue has been closed on Symfony side, this commit has been
edited and is now a simple line :tada:

Nek- commented:
I am sorry I do not understand why psalm do not pass on the CI and I can't suppress issue with ECS... I really do not understand how this package work. It seems like it should work with the following configuration, but without having it as a dependency I can't understand how it could work.

```php
<?php

declare(strict_types=1);

use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $containerConfigurator->import(__DIR__ . '/vendor/setono/coding-standard/easy-coding-standard.php');
};
```

☝️ missing on this package to allow ECS to work? But how can it work without setono/coding-standard as dependency ? But this package conflicts with sylius/coding-standard ?!

Nek- commented:
My hotfix can be removed: https://github.com/symfony/symfony/pull/43801

Nek- committed:
feat: make ecs config work

This commit also fixes some CS issues.
Please notice that this configuration is coming from the setono feed
bundle.

Nek- committed:
fix: psalm test suite on CI

This commit fixes the following problems:
1. The script inside composer.json was failing because on Github Action
   it runs with SH and not BASH (fixed by making it external)
2. Since we run Symfony 5.x, we need to adapt the psalm configuration of
   the Symfony plugin.

Nek- commented:
@loevgaard I SEE GREEN. 🙈  (finally)

_commits are all annotated with details_

loevgaard commented on src/DependencyInjection/Configuration.php:
```diff
@@ -20,16 +20,14 @@ public function getConfigTreeBuilder(): TreeBuilder
-            ->end()
-        ;
+                ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->cannotBeEmpty()
+        ; // Note: `->end()->end()` are removed since they do nothing and are dead code detected by psalm
```
Just remove these lines :)

Nek- replied:
Removed!

loevgaard commented on src/DependencyInjection/Configuration.php:
```diff
@@ -55,14 +53,15 @@ private function addResourcesSection(ArrayNodeDefinition $node): void
-                ->end()
-            ->end()
+                                        ->scalarNode('form')->defaultValue(PropertyType::class)->cannotBeEmpty() // ->end()
+//                                    ->end()
+//                                ->end()
+//                            ->end()
+//                        ->end()
+//                    ->end()
+//                ->end()
+//            ->end()
+        // statements `->end()` is dead code correctly detected by psalm
```
Just remove these lines :)

Nek- replied:
Removed!

loevgaard commented on src/EventListener/AddLibrarySubscriber.php:
```diff
@@ -23,7 +23,7 @@ public function add(RequestEvent $event): void
         $request = $event->getRequest();
 
-        if (!$event->isMasterRequest() || !$this->isShopContext($request)) {
+        if (!$this->isMainRequest($event) || !$this->isShopContext($request)) {
```
Doesn't work with SF4.4. You should revert these

Nek- replied:
It does! You probably misread it, it's a BC layer (by using a private method).

loevgaard commented on src/EventListener/TagSubscriber.php:
```diff
@@ -73,4 +74,15 @@ protected function isShopContext(Request $request = null): bool
+        }
+
+        // BC Layer for Symfony < 5.3
+        /** @psalm-suppress DeprecatedMethod */
```
Add it to Psalm config instead

Nek- replied:
How? (+ I actually think it's nice to have this comment to make it easy to drop the bc layer later)

loevgaard replied:
Yeah, I think you will gain the same benefits of just removing it from Psalm config, but let's just keep it :)

loevgaard commented on tests/Application/config/packages/lexik_jwt_authentication.yaml:
```diff
@@ -0,0 +1,4 @@
+lexik_jwt_authentication:
+    secret_key: '%env(resolve:JWT_SECRET_KEY)%'
+    public_key: '%env(resolve:JWT_PUBLIC_KEY)%'
+    pass_phrase: '%env(JWT_PASSPHRASE)%'
```
Not sure we need a password env var

Nek- replied:
Removed :) . (cute catch ^^)

loevgaard commented:
Just a few comments, else it's a good job 🎉

Nek- commented:
@loevgaard updated! :)

loevgaard commented:
Thank you, @Nek-. Great job 🎉

Nek- commented:
@loevgaard do you want me to forward commits to 3.x branch?

loevgaard commented:
@Nek-: `3.x` is basically a whole other plugin because we switched from client side to server side in that branch, but anything is welcome as always :)
